src/main/java/at/hannibal2/skyhanni/features/garden/visitor/VisitorRewardWarning.kt

```diff
@@ -0,0 +1,128 @@
+            return
+        }
+    }
+
+    @SubscribeEvent(priority = EventPriority.HIGH)
+    fun onStackClick(event: SlotClickEvent) {
+        if (!VisitorAPI.inInventory) return
+
+        val visitor = VisitorAPI.getVisitor(lastClickedNpc) ?: return
+        val blockReason = visitor.blockReason(event.slotId == ACCEPT_SLOT, event.slotId == REFUSE_SLOT)
+
+        if (blockReason == VisitorBlockReason.RARE_REWARD || blockReason == VisitorBlockReason.CHEAP_COPPER) {
+            event.isCanceled = true
+            return
+        }
+
+        if (blockReason == VisitorBlockReason.EXPENSIVE_COPPER) {
+            event.isCanceled = true
+            return
+        }
+
+        if (event.slotId == REFUSE_SLOT) {
+            if (event.slot.stack?.name != "§cRefuse Offer") return
+            VisitorAPI.changeStatus(visitor, VisitorAPI.VisitorStatus.REFUSED, "refused")
+            return
+        }
+        if (event.slotId == ACCEPT_SLOT) {
+            if (event.slot.stack?.name != "§eClick to give!") return
+            VisitorAPI.changeStatus(visitor, VisitorAPI.VisitorStatus.ACCEPTED, "accepted")
+            return
+        }
+    }
+
+    @SubscribeEvent(priority = EventPriority.HIGH)
+    fun onTooltip(event: ItemTooltipEvent) {
+        if (!GardenAPI.onBarnPlot) return
+        if (!VisitorAPI.inInventory) return
+        val visitor = VisitorAPI.getVisitor(lastClickedNpc) ?: return
+
+        val isRefuseSlot = event.itemStack.toString() == "1xtile.clayHardenedStained@14" // red
+        val isAcceptSlot = event.itemStack.toString() == "1xtile.clayHardenedStained@13" // green
+
+        val blockReason = visitor.blockReason(isAcceptSlot, isRefuseSlot)
+        if (blockReason == VisitorBlockReason.NONE) return
+
+        val copiedTooltip = event.toolTip.toList()
+        event.toolTip.clear()
+
+        for (line in copiedTooltip) {
+            if (line.contains("§aAccept Offer§r")) {
+                event.toolTip.add(line.replace("§aAccept Offer§r", "§7Accept Offer§8"))
+            }
+            else if (line.contains("§cRefuse Offer§r")) {
+                event.toolTip.add(line.replace("§cRefuse Offer§r", "§7Refuse Offer§8"))
+            }
+            else if (!line.contains("minecraft:") && !line.contains("NBT:")) {
+                event.toolTip.add("§8" + line.removeColor())
+            }
+        }
+        event.toolTip.add("")
+        event.toolTip.add(blockReason.description)
+        event.toolTip.add("  §7(Bypass by holding ${Keyboard.getKeyName(config.bypassKey)})")
+    }
+
+    fun VisitorAPI.Visitor.blockReason(
+        isAcceptSlot: Boolean,
+        isRefuseSlot: Boolean,
+        ignoreBypassKey: Boolean = false
+    ): VisitorBlockReason {
+        if (config.bypassKey.isKeyHeld() && !ignoreBypassKey) return VisitorBlockReason.NONE
+        val visitorHasReward = this.hasReward() != null && config.preventRefusing
+        return when {
```

Comment:
change this to a if, else if, else. to increase readability as the line size is too long